### PR TITLE
[AKS] `az aks create/update`: Support in place param updates for managed prom

### PR DESCRIPTION
**Pull request by bragi92**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

```az aks update -n kaveeshcli22 -g kaveeshcli --enable-azure-monitor-metrics --ksm-metric-labels-allow-list "namespaces=[k8s-label-1,k8s-label-n]" --ksm-metric-annotations-allow-list "pods=[k8s-annotation-1,k8s-annotation-n]"```

**Description**
Updating the behavior of the az aks update command for managed prom to accept KSM parameter updates even when the addon is enabled on the cluster and stopping the error we throw saying that you have to disable the addon first before updating.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/addonconfiguration.py [hunk 1]
```diff
@@ -396,6 +396,7 @@ def update_addons(
 
 def add_ingress_appgw_addon_role_assignment(result, cmd):
     service_principal_msi_id = None
+    is_service_principal = False
     # Check if service principal exists, if it does, assign permissions to service principal
     # Else, provide permissions to MSI
     is_service_principal = False
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/addonconfiguration.py:402]: duplicated with line 402?

File: src/aks-preview/azext_aks_preview/azuremonitormetrics/azuremonitorprofile.py [hunk 2]
```diff
@@ -81,17 +81,22 @@ def ensure_azure_monitor_profile_prerequisites(
...
-            cluster_subscription,
-            cluster_resource_group_name,
-            cluster_name,
-            cluster_region,
-            raw_parameters,
-            create_flow
-        )
+            is_prometheus_enabled = check_azuremonitormetrics_profile(cmd, cluster_subscription, cluster_resource_group_name, cluster_name)
+        if is_prometheus_enabled:
+            print("Azure Prometheus is already enabled : This command will only allow updates to the KSM parameters. All other parameters will be ignored")
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/azuremonitormetrics/azuremonitorprofile.py:89]: It is better to use a logger if possible.

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=99890179&view=results) to validate the change.

- test_aks_create_with_azuremonitormetrics
- test_aks_update_with_azuremonitormetrics

File: src/aks-preview/azext_aks_preview/addonconfiguration.py (hunk 1 above)
Comment (bragi92) [reply src/aks-preview/azext_aks_preview/addonconfiguration.py]: Fixed.

File: src/aks-preview/azext_aks_preview/azuremonitormetrics/azuremonitorprofile.py (hunk 2 above)
Comment (bragi92) [reply src/aks-preview/azext_aks_preview/azuremonitormetrics/azuremonitorprofile.py]: Updated to use logger.

**Comment by bragi92** [comment]
Queued another [live_test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=99952508&view=results)

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**